web/views.py

```
@@ -7103,3 +7128,112 @@ def users_list(request: HttpRequest) -> HttpResponse:
+    try:
+        topic = ForumTopic.objects.get(pk=pk)
+        vote_type = request.POST.get("vote_type")
+
+        if vote_type not in ["up", "down"]:
+            # For form submissions, redirect back with an error message if needed
+            messages.error(request, "Invalid vote type")
+            return redirect("view_topic", pk=topic.id)
+
+        # Check if user already voted on this topic
+        vote, created = ForumVote.objects.get_or_create(
+            user=request.user, topic=topic, defaults={"vote_type": vote_type}
```

Ruff (0.8.2)

Trailing comma missing

Add trailing comma

(COM812)